src/shared/services/bolt/boltMappings.test.js

```diff
@@ -367,4 +368,221 @@ describe('boltMappings', () => {
+      expect(typedNumber instanceof neo4j.Integer).toEqual(true)
+    })
+
+    test('should apply node type', () => {
+      const rawNode = {
+        labels: ['Test'],
+        properties: [],
+        identity: { low: 5, high: 0 }
+      }
+
+      const typedNode = applyGraphTypes(rawNode)
+      expect(typedNode instanceof neo4j.types.Node).toEqual(true)
+      expect(typedNode.identity instanceof neo4j.Integer).toEqual(true)
+    })
+
+    test('should apply node type to array of data', () => {
+      const rawNodes = [{
+        labels: ['Test'],
+        properties: [],
+        identity: { low: 5, high: 0 }
+      }, {
+        labels: ['Test2'],
+        properties: [],
+        identity: { low: 15, high: 0 }
+      }]
+
+      const typedNodes = applyGraphTypes(rawNodes)
+      expect(typedNodes.length).toEqual(2)
+      expect(typedNodes[0] instanceof neo4j.types.Node).toEqual(true)
+      expect(typedNodes[0].identity instanceof neo4j.Integer).toEqual(true)
+      expect(typedNodes[1] instanceof neo4j.types.Node).toEqual(true)
+      expect(typedNodes[1].identity instanceof neo4j.Integer).toEqual(true)
+    })
+
+    test('should apply relationship type', () => {
+      const rawRelationship = {
+        type: 'TestType',
+        properties: [],
+        identity: { low: 1, high: 0 },
+        start: { low: 5, high: 0 },
+        end: { low: 10, high: 0 }
+      }
+
+      const typedRelationship = applyGraphTypes(rawRelationship)
+      expect(typedRelationship instanceof neo4j.types.Relationship).toEqual(true)
+      expect(typedRelationship.identity instanceof neo4j.Integer).toEqual(true)
+    })
+
+    test('should apply relationship type to array of data', () => {
+      const rawRelationships = [{
+        type: 'TestType',
+        properties: [],
+        identity: { low: 1, high: 0 },
+        start: { low: 5, high: 0 },
+        end: { low: 10, high: 0 }
+      }, {
+        type: 'TestType_2',
+        properties: [],
+        identity: { low: 2, high: 0 },
+        start: { low: 15, high: 0 },
+        end: { low: 20, high: 0 }
+      }]
+
+      const typedRelationships = applyGraphTypes(rawRelationships)
+      expect(typedRelationships.length).toEqual(2)
+      expect(typedRelationships[0] instanceof neo4j.types.Relationship).toEqual(true)
+      expect(typedRelationships[0].identity instanceof neo4j.Integer).toEqual(true)
+      expect(typedRelationships[0].start instanceof neo4j.Integer).toEqual(true)
+      expect(typedRelationships[0].end instanceof neo4j.Integer).toEqual(true)
+      expect(typedRelationships[1] instanceof neo4j.types.Relationship).toEqual(true)
+      expect(typedRelationships[1].identity instanceof neo4j.Integer).toEqual(true)
+      expect(typedRelationships[1].start instanceof neo4j.Integer).toEqual(true)
+      expect(typedRelationships[1].end instanceof neo4j.Integer).toEqual(true)
+    })
+
+    test('should apply to custom object properties', () => {
```

Comment:
This test tests nodes again, not regular js objects. Please add tests for primitives as well, like `bool`, `null`, `string`, `float` etc.